test/test_pqueue.c

```diff
@@ -31,8 +32,9 @@ void test_pqueue_desc_order(void) {
 void test_pqueue_out_of_space(void) {
   // We're using a 1-indexed minheap, so only expect success up to (# nodes - 1)
   for (int i = 0; i < TEST_PQUEUE_SIZE + 5; i++) {
-    bool result = pqueue_push(&gv_queue, i, i);
-    TEST_ASSERT_EQUAL(((i + 1) < TEST_PQUEUE_SIZE), result);
+    StatusCode result = pqueue_push(&gv_queue, i, i);
+    TEST_ASSERT_EQUAL(
+        ((i + 1) < TEST_PQUEUE_SIZE) ? STATUS_CODE_OK : STATUS_CODE_RESOURCE_EXHAUSTED, result);
```

Comment:
We should move this to a new line or make it an explicit if statement. I'm not a huge fan of complex ternary operations.